File: mobile/lib/domain/models/journal.dart
class JournalEntry {
  final String id;
  final String title;
  final String? problem;
  final String? rootCause;
  final String? solution;
  final String? conceptLearned;
  final String? codeSnippet;
  final String? language;
  final String? projectName;
  final bool isFavorite;
  final DateTime createdAt;
  final DateTime updatedAt;

  JournalEntry({
    required this.id,
    required this.title,
    this.problem,
    this.rootCause,
    this.solution,
    this.conceptLearned,
    this.codeSnippet,
    this.language,
    this.projectName,
    this.isFavorite = false,
    required this.createdAt,
    required this.updatedAt,
  });

  factory JournalEntry.fromJson(Map<String, dynamic> json) {
    return JournalEntry(
      id: json['id'],
      title: json['title'],
      problem: json['problem'],
      rootCause: json['rootCause'],
      solution: json['solution'],
      conceptLearned: json['conceptLearned'],
      codeSnippet: json['codeSnippet'],
      language: json['language'],
      projectName: json['projectName'],
      isFavorite: json['isFavorite'] ?? false,
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
    );
  }
}

File: backend/src/db/schema/journal.ts
import { pgTable, uuid, varchar, text, timestamp, boolean, index } from "drizzle-orm/pg-core";
import { users } from "./users";

export const journalEntries = pgTable("journal_entries", {
    id: uuid("id").primaryKey().defaultRandom(),
    userId: uuid("user_id").references(() => users.id, { onDelete: "cascade" }).notNull(),
    title: varchar("title", { length: 255 }).notNull(),
    problem: text("problem"),
    rootCause: text("root_cause"),
    solution: text("solution"),
    conceptLearned: text("concept_learned"),
    codeSnippet: text("code_snippet"),
    language: varchar("language", { length: 50 }),
    projectName: varchar("project_name", { length: 100 }),
    isFavorite: boolean("is_favorite").default(false),
    createdAt: timestamp("created_at", { withTimezone: true }).defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).defaultNow(),
}, (t) => [
    index("journal_entries_user_id_idx").on(t.userId),
]);

export const journalTags = pgTable("journal_tags", {
    id: uuid("id").primaryKey().defaultRandom(),
    userId: uuid("user_id").references(() => users.id, { onDelete: "cascade" }).notNull(),
    name: varchar("name", { length: 50 }).notNull(),
    color: varchar("color", { length: 7 }).default("#6366F1"),
    createdAt: timestamp("created_at", { withTimezone: true }).defaultNow(),
}, (t) => [
    index("journal_tags_user_id_idx").on(t.userId),
]);

export const journalEntryTags = pgTable("journal_entry_tags", {
    journalId: uuid("journal_id").references(() => journalEntries.id, { onDelete: "cascade" }).notNull(),
    tagId: uuid("tag_id").references(() => journalTags.id, { onDelete: "cascade" }).notNull(),
}, (table) => ({
    pk: { primaryKey: { columns: [table.journalId, table.tagId] } },
}));